test/specs/progress.spec.js

```diff
@@ -0,0 +1,94 @@
+describe('progress events', function () {
+  beforeEach(function () {
+    jasmine.Ajax.install();
+  });
+
+  afterEach(function () {
+    jasmine.Ajax.uninstall();
+  });
+
+  it('should add a download progress handler', function (done) {
+    function progressHandler(event) {};
+
+    axios('/foo', { progressDownload: progressHandler } );
+
+    getAjaxRequest().then(function (request) {
+      // Not sure of the best way to test this, so just test that the new listener is present
+      expect(request.eventBus.eventList.progress.length).toEqual(2);
```

Comment:
Yeah i agree, i don't like that at all on a second look!
Plus the upload test doesn't actually test for anything atm as i couldn't figure out the best way to test it as Jasmine AJAX doesn't seem to catch/handle upload events.

I'll have another dive in and see how i can improve both. :)